File: datacontainer/code/dynostore/utils/csvlog.py
# dynostore/utils/csvlog.py
import logging

def make_csv_logger(service: str, logger_name: str | None = None):
    """
    Returns a _log(operation, key, phase, status, msg="", level="debug") function
    that emits: SERVICE, OPERATION, OBJECTKEY, START/END, Status, MSG
    """
    logger = logging.getLogger(logger_name or service.lower())

    def _log(operation: str, key: str, phase: str, status: str, msg: str = "", level: str = "debug"):
        rec = f"{service},{operation},{key},{phase},{status},{msg}"
        lvl = level.lower()
        if   lvl == "info":     logger.info(rec)
        elif lvl == "warning":  logger.warning(rec)
        elif lvl == "error":    logger.error(rec)
        elif lvl == "critical": logger.critical(rec)
        else:                   logger.debug(rec)   # default
    return _log

File: metadata/app/app/Support/csv_log.php
<?php

use Illuminate\Support\Facades\Log;

if (!function_exists('csv_log')) {
    /**
     * Emit: SERVICE,OPERATION,OBJECTKEY,START/END,Status,MSG
     * @param string $level debug|info|warning|error|critical
     */
    function csv_log(
        string $service,
        string $operation,
        string $objectKey,
        string $phase,
        string $status,
        string $msg = '',
        string $level = 'debug'
    ): void {
        $line = "{$service},{$operation},{$objectKey},{$phase},{$status},{$msg}";
        Log::channel('csv')->{$level}($line);
    }
}
